src/meshlink.c

```diff
@@ -2656,9 +2687,21 @@ void meshlink_whitelist(meshlink_handle_t *mesh, meshlink_node_t *node) {
 	pthread_mutex_lock(&(mesh->mesh_mutex));
 
 	node_t *n = (node_t *)node;
+
+	if(!n->status.blacklisted) {
+		logger(mesh, MESHLINK_DEBUG, "Node %s was never blacklisted\n", node->name);
```

Comment:
I would make the error mesage "Node %s was already whitelisted", because "never blacklisted" might not actually be true.

Also, if `n == mesh->self`, then this should also set `errno = MESHLINK_EINVAL`, just like with `meshlink_blacklist()`.